gimie/make_graph.py

```diff
@@ -0,0 +1,75 @@
+from rdflib import Graph, URIRef, parser
+import hashlib
+from calamus import fields
+from calamus.schema import JsonLDSchema
+
+schema = fields.Namespace("http://schema.org/")
+
+
+def generate_fair_uri(
+    repository_path: str = "https://example.com/Repository/",
+):
+    """given a repository_path, returns a URI with a hash for uniqueness, or the repository URL if it's online"""
+    # Compute the SHA-256 hash of the repository name
+    hash = hashlib.sha256(repository_path.encode()).hexdigest()
+    # TODO https://www.github.com/repoX should return the same result as github.com/repoX
+    if (
+        False
+    ):  # repository_path.is_online(): #we need a method that checks whether the repopath is online or a foldername
+        fair_uri = repository_path
+    else:
+        # Return the FAIR URI in the form "fair:sha256-HASH, truncated to 5 characters to promote readability"
+        fair_uri = (
+            f"gimie:{repository_path}/" + hash[:5]
+        )  # TODO decide on URI we want to use for non online repos
+    return fair_uri
+
+
+class Repository:
```

Comment:
This will need to be discussed, but I think this should either be:
* _The_ user-facing class. if so, it will probably need to live elsewhere in the API and be renamed since `gimie.repo.Repo` is already a thing.
* or a wrapper class used only for graph generation (e.g. `ProjectGraph`).

Reply:
Maybe for now we can just rename it to ProjectGraph and use it as a temporary placeholder

Reply:
Agreed!